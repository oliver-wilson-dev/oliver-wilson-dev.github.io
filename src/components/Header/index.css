.header {
    top: 0;
    left: 0;
    z-index: 6;
    width: 100%;
    display: flex;
    position: fixed;
    align-items: center;
    box-sizing: border-box;
    justify-content: center;
    background-color: var(--primaryBackgroundLight);
    height: var(--headerHeight);
    padding: var(--headerPadding);
    box-shadow:  0px 0px 8px var(--primaryBorderColourLight);
}

.wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    max-width: 850px;
    align-items: center;
    justify-content: space-between;
}

.icon {
    width: 30px;
    height: 30px;
}

.burgerMenu {
    @extend .icon;
}

.crossIcon {
    @extend .icon;
    position: absolute;
    top: 0;
    right: 0;
    margin: 10px;
    margin: var(--headerPadding);
    fill: var(--primaryTextColor);
    opacity: 0;

    transition-property: opacity;
    transition-duration: .3s;
    transition-timing-function: ease;
}

.crossIconShow {
    opacity: 1;
}

.menuContainer {
    width: 0px;
    background-color: transparent;

    height: 100vh;
    z-index: 5;
    position: absolute;
    top: 0;
    right: 0;

    display: flex;
    justify-content: flex-end;
}

@keyframes open {
    0% {
        width: 0px;
        background-color: transparent;
    }

    50% {
        width: 100vw;
        background-color: transparent;
    }

    100% {
        width: 100vw;
        background-color: var(--overlayBackground);
    }
}

@keyframes close {
    0% {
        width: 100vw;
        background-color: var(--overlayBackground);
    }

    50% {
        width: 100vw;
        background-color: transparent;
    }

    100% {
        width: 0px;
        background-color: transparent;
    }
}

.menuOpen {
    animation-name: open;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-direction: normal;
}

.menuClose {
    animation-name: close;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-direction: normal
}

.menuHideFirstRender {
    display: none;
}

.menuContent {
    width: 80%;
    display: flex;
    height: 100%;
    background-color: var(--primaryBackground);
    box-shadow: 0px 0px 8px var(--primaryBorderColourLight);
}

.menuList {
    width: 100%;
    padding: 0px;
    margin: 20px 0px;
    list-style-type: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.menuItem {
    text-align: center;
    text-align: center;
    font-weight: 700;
    font-size: 30px;
    font-family: 'Source Sans Pro', sans-serif;
    padding: 40px;

    &:hover {
        background-color: var(--primaryBackgroundInverse);
        color: var(--primaryTextColorInverse);
    }
}

.link {
    color: var(--primaryTextColor);
}

@media (min-width: 786px) {
    .menuContent {
        max-width: 600px;
    }
}