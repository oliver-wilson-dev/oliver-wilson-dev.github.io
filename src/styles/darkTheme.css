@import "./index.css";

[data-theme="DARK_THEME"] {
    --primaryBackground: #333;
    --primaryBackgroundImage: linear-gradient(#333, #333);
    --primaryBorderColour: #f8f8f8;
    --primaryTextColor: #f8f8f8;
    --spinnerTrailColour: rgba(248,248,248, 0.2);
    --sectionFadeColourFrom: rgba(51, 51, 51, 0);
    --sectionFadeColourTo: rgba(51, 51, 51, 1);
    --unvisitedLink: var(--projectGreen)
}