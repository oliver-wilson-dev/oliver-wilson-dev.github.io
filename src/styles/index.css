:root {
    --primaryBackgroundLight: #f8f8f8;
    --primaryBackgroundDark: #333;
    --primaryBackground: var(--primaryBackgroundLight);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: #333;

    --primaryTextColorDark: #f8f8f8;
    --primaryTextColorLight: #333;

    --primaryTextColorInverse: var(--primaryTextColorLight);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --linkBlue: rgb(21, 64, 204);
    --unvisitedLink: var(--linkBlue);

    --profilePictureBorder: var(--primaryBorderColour);

    --headerHeight: 65px;
    --headerPadding: 15px;

    --overlayBackground: rgba(0,0,0,.288);

    --projectGreen: #06d6a0;
    --projectPink: #EF476F;
    --projectYellow: #FFD166;
    --projectBlue: #118AB2;
    --projectDarkBlue: #073B4C;
}