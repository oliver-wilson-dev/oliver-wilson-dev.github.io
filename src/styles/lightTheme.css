@import "./index.css";

[data-theme="LIGHT_THEME"] {
    --primaryBackground: var(--primaryBackgroundLight);
    --primaryBackgroundInverse: var(--primaryBackgroundDark);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: var(--primaryTextColorLight);
    --primaryTextColorInverse: var(--primaryTextColorDark);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --primaryBackgroundLight: #f8f8f8;

    --unvisitedLink: var(--projectGreen);
    --unvisitedLinkInverse: var(--projectGreen);
}