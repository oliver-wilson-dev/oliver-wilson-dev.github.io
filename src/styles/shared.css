.bodyFixed {
    position: fixed;
    left: 0;
    width: 100%;
    overflow-y: scroll;
}

.pageHeader {
    font-size: 5em;
    margin: 0;
    margin-top: 10px;
    background-clip: text;
    -webkit-background-clip: text;
    letter-spacing: 2px;
    text-align: center;
    width: 100%;
  }

  @media (min-width: 786px) {
    .landing--header {
    max-width: 800px;
    }
  }

  @media (min-width: 0px) and (max-width: 767px) {
    .pageHeader {
      font-size: 3em;
    }
  }