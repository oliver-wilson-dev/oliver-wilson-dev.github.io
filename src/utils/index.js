export { preventScroll, allowScroll } from './overlayScroll';
