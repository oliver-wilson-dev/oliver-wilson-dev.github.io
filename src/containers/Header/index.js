import { connect } from 'react-redux';
import Header from '../../components/Header';
import { getShowSideNav } from '../../state/selectors';
import toggleShowSideNav from '../../state/actions/toggleShowSideNav';

const mapStateToProps = state => ({
  showSideNav: getShowSideNav(state)
});

const mapDispatchToProps = dispatch => ({
  toggleShowSideNav: () => dispatch(toggleShowSideNav())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
