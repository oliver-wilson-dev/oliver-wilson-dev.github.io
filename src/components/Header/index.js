import React, { useState, useRef, useEffect } from 'react';
import cn from 'classnames';
import styles from './index.css';
import ThemeToggle from '../../containers/ThemeToggle';
import BurgerMenu from './burgerMenu.svg';
import CrossIcon from '../Icons/cross.svg';
import { preventScroll, allowScroll } from '../../utils';

const Header = () => {
  const [open, setOpen] = useState(false);
  const [close, setClose] = useState(false);
  const isFirstRender = useRef(true);

  const toggleMenu = () => {
    const burgerMenuNotYetClicked = !open && !close;

    if (burgerMenuNotYetClicked) {
      setOpen(true);
      setClose(false);
    } else {
      setOpen(!open);
      setClose(!close);
    }
  };

  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
    } else if (open) {
      preventScroll();
    } else { allowScroll(); }
  }, [open]);

  return (
    <nav
      className={cn(styles.header)}
    >
      <div className={cn(styles.wrapper)}>
        <ThemeToggle />
        <BurgerMenu className={styles.burgerMenu} onClick={toggleMenu} />
      </div>
      <div
        className={cn(styles.menuContainer, {
          [styles.menuOpen]: open,
          [styles.menuClose]: close && !open,
          [styles.menuHideFirstRender]: isFirstRender.current
        })}
      >
        <div
          className={styles.menuContent}
        >
          <CrossIcon
            className={cn(styles.crossIcon, {
              [styles.crossIconShow]: open
            })}
            onClick={toggleMenu}
          />
          <ul className={styles.menuList}>
            <li className={styles.menuItem}>
              Home
            </li>
            <li className={styles.menuItem}>
              About
            </li>
            <li className={styles.menuItem}>
              Education
            </li>
            <li className={styles.menuItem}>
              Experience
            </li>
          </ul>
        </div>


      </div>
    </nav>
  );
};


export default Header;
