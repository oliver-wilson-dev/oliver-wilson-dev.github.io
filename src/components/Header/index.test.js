import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Header from './index';
import styles from './index.css';
import { preventScroll, allowScroll } from '../../utils';

jest.mock('../../utils', () => ({
  preventScroll: jest.fn(),
  allowScroll: jest.fn()
}));

jest.mock('../Icons/cross.svg', () => {
  const CrossIcon = () => null;

  return CrossIcon;
});

jest.mock('./burgerMenu.svg', () => {
  const BurgerMenu = () => null;

  return BurgerMenu;
});

jest.mock('../../containers/ThemeToggle', () => {
  const ThemeToggle = () => null;

  return ThemeToggle;
});

const defaultProps = { theme: 'test-theme' };

const render = (
  overrideProps,
  renderMethod = shallow
) => renderMethod(<Header {...defaultProps} {...overrideProps} />);

describe('<Header/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });


  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when clicking the burgerMenu icon', () => {
    it('should open the sub navigation', () => {
      const component = render();

      component.find(`.${styles.burgerMenu}`).simulate('click');

      expect(component).toMatchSnapshot();
    });

    it('should call preventScroll', () => {
      const component = render({}, mount);

      act(() => {
        component.find(`.${styles.burgerMenu}`).props().onClick();
      });

      expect(preventScroll).toHaveBeenCalledTimes(1);
    });

    describe('and then clicking the close icon', () => {
      it('should call allowScroll', () => {
        const component = render({}, mount);

        // can't click on the svg element in jest for some reason
        act(() => {
          component.find(`.${styles.burgerMenu}`).props().onClick();
        });

        component.update();

        act(() => {
          component.find(`.${styles.crossIcon}`).props().onClick();
        });

        expect(allowScroll).toHaveBeenCalledTimes(1);
      });
    });
  });
});
