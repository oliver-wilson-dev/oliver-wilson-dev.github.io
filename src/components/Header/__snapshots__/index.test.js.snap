// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> component should render correctly 1`] = `
<nav
  className="header"
>
  <div
    className="wrapper"
  >
    <ThemeToggle />
    <BurgerMenu
      className="burgerMenu"
      onClick={[Function]}
    />
  </div>
  <div
    className="menuContainer menuHideFirstRender"
  >
    <div
      className="menuContent"
    >
      <CrossIcon
        className="crossIcon"
        onClick={[Function]}
      />
      <ul
        className="menuList"
      >
        <li
          className="menuItem"
        >
          Home
        </li>
        <li
          className="menuItem"
        >
          About
        </li>
        <li
          className="menuItem"
        >
          Education
        </li>
        <li
          className="menuItem"
        >
          Experience
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`<Header/> component when clicking the burgerMenu icon should open the sub navigation 1`] = `
<nav
  className="header"
>
  <div
    className="wrapper"
  >
    <ThemeToggle />
    <BurgerMenu
      className="burgerMenu"
      onClick={[Function]}
    />
  </div>
  <div
    className="menuContainer menuOpen menuHideFirstRender"
  >
    <div
      className="menuContent"
    >
      <CrossIcon
        className="crossIcon crossIconShow"
        onClick={[Function]}
      />
      <ul
        className="menuList"
      >
        <li
          className="menuItem"
        >
          Home
        </li>
        <li
          className="menuItem"
        >
          About
        </li>
        <li
          className="menuItem"
        >
          Education
        </li>
        <li
          className="menuItem"
        >
          Experience
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
